samcli/local/common/runtime_template.py

```diff
@@ -83,11 +83,15 @@ def get_local_lambda_images_location(mapping, runtime):
     return os.path.join(_lambda_images_templates, runtime, dir_name + "-lambda-image")
 
 
-SUPPORTED_DEP_MANAGERS: Set[str] = {
-    c["dependency_manager"]  # type: ignore
-    for c in list(itertools.chain(*(RUNTIME_DEP_TEMPLATE_MAPPING.values())))
-    if c["dependency_manager"]
-}
+SUPPORTED_DEP_MANAGERS: List[str] = list(
+    set(
+        {
+            c["dependency_manager"]  # type: ignore
+            for c in list(itertools.chain(*(RUNTIME_DEP_TEMPLATE_MAPPING.values())))
+            if c["dependency_manager"]
```

Comment:
should we use methods like c.get() which seems safer in this context?

Reply:
Not critical as this would be caught at build time if `c["dependency_manager"]` throws a `KeyError` but generally a good practice. Not sure why we wrote it this way to begin with but updated to your suggestion.